src/common/libutil/test/fdwalk.c

```diff
@@ -46,14 +46,22 @@ static void set_fd (void *data, int fd)
     fds[fd]++;
 }
 
+static void set_fd_if_open (void *data, int fd)
+{
+    if (fcntl (fd, F_GETFL) < 0 && errno == EBADF)
+        return;
+    int *fds = data;
+    fds[fd]++;
+}
+
 static int * get_open_fds (int maxfd)
 {
     /* Valgrind may show open fds > maxfd, so double the space
      *  allocated so we don't overflow when run under valgrind.
      */
     int * fds = calloc (maxfd * 2, sizeof (int));
     if (fds)
-        ok (fdwalk (set_fd, fds) == 0,
+        ok (fdwalk (set_fd_if_open, fds) == 0,
```

Comment:
should we also do this in `test_fdwalk_fallback()`?

Reply:
I'm not seeing much motivation for doing that, are you?

Reply:
just wanted to make sure it wasn't missed by accident :-)